index.d.ts

```diff
@@ -0,0 +1,14 @@
+declare module '@metamask/swappable-obj-proxy' {
+  export type SwappableProxy<T = unknown> = T & {
+    setTarget: (newTarget: T) => void,
+  };
+
+  export function createEventEmitterProxy<T = unknown>(
```

Comment:
Nit: Should we require that this type extends `EventEmitter`? It does assume the target has EventEmitter methods, and it invokes them.

Reply:
Oh, yes, I suppose you're right. I also found that the type for `initialTarget` was wrong. Fixed in [9835937](https://github.com/MetaMask/swappable-obj-proxy/pull/13/commits/9835937634e4d9f76e28e0768596430959578233).